modules/data.atmosphere/R/download.Drought2018.R

```diff
@@ -0,0 +1,195 @@
+      	VALUES ?station {<http://meta.icos-cp.eu/resources/stations/ES_sitename>}
+      			?dobj cpmeta:wasAcquiredBy/prov:wasAssociatedWith ?station .
+      ?dobj cpmeta:hasStartTime | (cpmeta:wasAcquiredBy / prov:startedAtTime) ?timeStart .
+      	FILTER NOT EXISTS {[] cpmeta:isNextVersionOf ?dobj}
+      }
+      "
+      body <- gsub("sitename", sitename, body)
+      response <- httr::POST(url, body = body)
+      response <- httr::content(response, as = "text")
+      response <- jsonlite::fromJSON(response)
+      dataset_url <- response$results$bindings$dobj$value
+      dataset_start_date <-
+        as.Date(
+          strptime(response$results$bindings$timeStart$value, format = "%Y-%m-%dT%H:%M:%S")
+        )
+      if (is.null(dataset_url)) {
+        PEcAn.logger::logger.severe("Data is not available for the requested site")
+      }
+      if (dataset_start_date > as.Date(start_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested start date. Please try again with",
+            dataset_start_date,
+            "as start date."
+          )
+        )
+      }
+      dataset_id <- sub(".*/", "", dataset_url)
+      
+      # construct the download URL
+      download_url <-
+        paste0('https://data.icos-cp.eu/licence_accept?ids=%5B%22',
+               dataset_id,
+               '%22%5D')
+      # Download the zip file
+      file <-
+        httr::GET(url = download_url,
+                  httr::write_disk(
+                    paste0(outfolder, "/Drought", sitename, ".zip"),
+                    overwrite = TRUE
+                  ),
+                  httr::progress())
+    }
+    
+    if (stage_extract) {
+      file_name <- paste0('FLX_', sitename, '_FLUXNET2015_FULLSET_HH')
+      
+      # extract only the hourly data file
+      zipped_csv_name <-
+        grep(
+          paste0('^', file_name),
+          utils::unzip(zip_file_name, list = TRUE)$Name,
+          ignore.case = TRUE,
+          value = TRUE
+        )
+      utils::unzip(file.path(outfolder, paste0('Drought', sitename, '.zip')),
+                   files = zipped_csv_name,
+                   exdir = outfolder)
+      `%>%` <- dplyr::`%>%`
+      # read in the output CSV file and select the variables required
+      df <- utils::read.csv(file.path(outfolder, zipped_csv_name))
+      df <-
+        subset(
+          df,
+          select = c(
+            "TIMESTAMP_START",
+            "TA_F",
+            "SW_IN_F",
+            "LW_IN_F",
+            "VPD_F",
+            "PA_F",
+            "P_F",
+            "WS_F",
+            "WD",
+            "RH",
+            "PPFD_IN",
+            "CO2_F_MDS",
+            "TS_F_MDS_1",
+            "TS_F_MDS_2",
+            "NEE_VUT_REF",
+            "LE_F_MDS",
+            "RECO_NT_VUT_REF",
+            "GPP_NT_VUT_REF"
+          )
+        )
+      df <-
+        df %>% dplyr::filter((
```

Comment:
Should I not subset the variables either (i.e include all the 233 variables in the output file)? 

> E.g. if end year in your example was 2017, here you would subset it until 2017 but write it to a "FLX_FI-Sii_FLUXNET2015_FULLSET_HH_2016-2018_beta-3.csv" file, right?

Yes, I just realized I didn't change the end date in the file name to reflect the actual end date.

Reply:
hmm I was only talking about the date-subsetting but I'm now wondering if we should keep all the variables in the csv as well. It would make files larger, but if we follow Ameriflux example we should keep the raw csv file as it is and then define columns in the format with respect to the full csv column numbers